pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -873,12 +873,16 @@ public synchronized PulsarClient getClient() throws PulsarServerException {
                 ClientBuilder builder = PulsarClient.builder()
                     .serviceUrl(this.getConfiguration().isTlsEnabled()
                                 ? this.brokerServiceUrlTls : this.brokerServiceUrl)
-                    .enableTls(this.getConfiguration().isTlsEnabled())
-                    .allowTlsInsecureConnection(this.getConfiguration().isTlsAllowInsecureConnection())
-                    .tlsTrustCertsFilePath(this.getConfiguration().getTlsCertificateFilePath());
+                    .enableTls(this.getConfiguration().isTlsEnabled());
+
+                if (this.getConfiguration().isBrokerClientTlsEnabled()) {
+                    builder.allowTlsInsecureConnection(this.getConfiguration().isTlsAllowInsecureConnection());
+                    builder.tlsTrustCertsFilePath(this.getConfiguration().getBrokerClientTrustCertsFilePath());
```

Comment:
Generally the cacert should be the same. What scenario let you to this problem?

Reply:
This is internal client, and the setting should be `brokerClientXXX`

Reply:
yes, I get that. I'm just wondering what sort of set up requires a different CA cert for broker client and for the server. The broker client is going to connect to other brokers in the same cluster, which will have the same CA. 

My concern here is that people will have configured with only the tls certificate path, and not the broker client tls cert path. This change will break their configuration. Given than both paths are likely to be the same, if broker client tls cert path is empty, it should fall back to tls certificate path.

Reply:
@ivankelly, Thanks. A user meet this issue in their environment. As you mentioned, the content of path for the 2 configs could be the same, but we should provide the ability for user to set it differently. All the auth logic is the same for internal client, in all the admin-client and the replicator-client, it is all set as this change, this is the only place that is not align.

Reply:
Providing the ability to have different values is good. However, current behaviour is that just setting tlsCertificateFilePath is enough for both. This change _requires_ that both values are set, which breaks people using the current behaviour. The correct fix here is to use brokerClientTrustCertsFilePath it is non-empty, otherwise use tlsCertificateFilePath.

Reply:
changed following your comments, and merged with latest master

Reply:
@ivankelly. thanks. changed following your comments, and merged with latest master